### ISSUE TITLE
`predict()` with weights fails when weighting variable has attributes?

### ISSUE THREAD

**Issue by strengejacke** [glmmTMB/glmmTMB#977]
Not sure if you can internally handle this, i.e. somehow drop the attributes of weight-variables within `predict()`?

``` r
library(glmmTMB)
data(iris)

d <- as.data.frame(expand.grid(
  Species = unique(iris$Species),
  Petal.Width = 2,
  wg = NA
))

iris$wg <- abs(rnorm(nrow(iris), 1, 0.1))

m <- glmmTMB(
  Petal.Length ~ Petal.Width + (1 | Species),
  weights = wg,
  data = iris
)
predict(m, newdata = d, re.form = NULL)
#> [1] 3.309128 4.970012 5.506580

attr(iris$wg, "label") <- "weighting variable"
m <- glmmTMB(
  Petal.Length ~ Petal.Width + (1 | Species),
  weights = wg,
  data = iris
)
predict(m, newdata = d, re.form = NULL)
#> Problem with these data entries:
#> weights 
#>      14
#> Error in MakeADFun(data.tmb, parameters, map = mapArg, random = randomArg, : Only numeric matrices, vectors, arrays, factors, lists or length-1-characters can be interfaced
```

<sup>Created on 2024-01-16 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by bbolker** [comment]
Maybe fixed with 726a705fabef190244bef321248d4fc756f79c09 ?

**Comment by strengejacke** [comment]
Thanks for quickly fixing this!
